torchvision/csrc/cuda/ROIPool_cuda.cu

```diff
@@ -116,8 +116,8 @@ std::tuple<at::Tensor, at::Tensor> ROIPool_forward_cuda(const at::Tensor& input,
   auto height = input.size(2);
   auto width = input.size(3);
 
-  at::Tensor output = input.type().tensor({num_rois, channels, pooled_height, pooled_width});
-  at::Tensor argmax = input.type().toScalarType(at::kInt).tensor({num_rois, channels, pooled_height, pooled_width}).zero_();
+  at::Tensor output = at::zeros({num_rois, channels, pooled_height, pooled_width}, input.type());
```

Comment:
shouldn't this be `input.options()` ?

Reply:
Yup. I've actually seen test cases of ATen where both are acceptable so I was kind of confused, but your suggestion is more general.

Reply:
I think `input.type()` is deprecated.

Reply:
Oh then what should be passed as the first argument to `AT_DISPATCH_FLOATING_TYPES_AND_HALF`?
I am currently passing in `input.type()` or `grad.type()` based on the pass.